lib/lrmd/lrmd_client.c

```diff
@@ -1403,6 +1408,28 @@ add_tls_to_mainloop(lrmd_t *lrmd, bool do_handshake)
     return rc;
 }
 
+static gboolean
+try_handshake_trigger(gpointer user_data)
+{
+    lrmd_t *lrmd = user_data;
+    lrmd_private_t *native = lrmd->lrmd_private;
+    pcmk__remote_t *remote = native->remote;
+    int rc = pcmk_rc_ok;
+    int tls_rc = GNUTLS_E_SUCCESS;
+
+    rc = pcmk__tls_client_handshake(remote, TLS_HANDSHAKE_TIMEOUT, &tls_rc);
+
+    /* Handshake completed, successfully or not.  Process that. */
+    if (rc == pcmk_rc_ok) {
+        tls_handshake_succeeded(lrmd);
+    } else {
+        report_async_connection_result(lrmd, -EKEYREJECTED);
+        tls_handshake_failed(lrmd, tls_rc, rc);
+    }
```

Comment:
We still want to do the try version here and reset the trigger on EAGAIN. We'll also need to track the timeout somehow so we don't do it infinitely.

Reply:
If we get to the else case here, we've called pcmk__tls_client_handshake -> pcmk__tls_client_try_handshake which has already dealt with EAGAIN and the timeout.

Reply:
right but that's not async :) that's part of what has to be changed

`pcmk__tls_client_handshake()` should only be called for sync connections now